examples/module/reprojpgo/reprojpgo.py

```diff
@@ -0,0 +1,103 @@
+import torch
+import argparse
+import pypose as pp
+from pathlib import Path
+import pypose.optim as ppopt
+from dataset import ReprojErrDataset, visualize, report_pose_error
+
+
+class ReprojectErrorGraph(torch.nn.Module):
+    def __init__(self, K, pts1, pts2, depth, init_T) -> None:
+        super().__init__()
+        self.register_buffer("K", K)
+        self.register_buffer("pts1", pts1)  # N x 2, uv coordinate
+        self.register_buffer("pts2", pts2)  # N x 2, uv coordinate
+        self.fx, self.fy = K[0, 0], K[1, 1]
+        self.cx, self.cy = K[0, 2], K[1, 2]
+
+        self.T = pp.Parameter(init_T)
+        self.depth = torch.nn.Parameter(depth)
+
+    def pts3d(self) -> torch.Tensor:
+        pts3d_z = self.depth
+        pts3d_x = ((self.pts1[..., 0] - self.cx) * pts3d_z) / self.fx
+        pts3d_y = ((self.pts1[..., 1] - self.cy) * pts3d_z) / self.fy
+        return torch.stack([pts3d_x, pts3d_y, pts3d_z], dim=1)  # Nx3
+
+    @torch.no_grad()
+    def reproject(self) -> torch.Tensor:
+        reproj_uv = pp.function.point2pixel(self.pts3d(), self.K, self.T.Inv())
+        return reproj_uv

```

Comment:
this is used only in visualization, I don't suggest putting it in models. how about putting it in the visualization function?